bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -0,0 +1,261 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.AbstractMap;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.jar.Manifest;
+
+import org.apache.maven.project.MavenProject;
+import org.bndtools.api.ILogger;
+import org.bndtools.api.Logger;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.IMavenProjectChangedListener;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.IMavenProjectRegistry;
+import org.eclipse.m2e.core.project.MavenProjectChangedEvent;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.bnd.osgi.Domain;
+import aQute.bnd.osgi.repository.BaseRepository;
+import aQute.bnd.service.Refreshable;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.bnd.version.Version;
+import aQute.lib.collections.SortedList;
+
+public class MavenWorkspaceRepository extends BaseRepository implements Repository, RepositoryPlugin, Refreshable, IMavenProjectChangedListener {
+
+    private final ILogger logger = Logger.getLogger(MavenWorkspaceRepository.class);
+
+    private final IMavenProjectRegistry mavenProjectRegistry = MavenPlugin.getMavenProjectRegistry();
+
+    private boolean inited = false;
+
+    private final Map<String,Entry<IMavenProjectFacade,MavenProject>> bsnMap = new HashMap<>();
+
+    @Override
+    public Map<Requirement,Collection<Capability>> findProviders(Collection< ? extends Requirement> requirements) {
+        return null;
+    }
+
+    @Override
+    public boolean refresh() throws Exception {
+        return false;
+    }
+
+    @Override
+    public File getRoot() throws Exception {
+        return null;
+    }
+
+    @Override
+    public PutResult put(InputStream stream, PutOptions options) throws Exception {
+        return null;
+    }
+
+    @Override
+    public File get(String bsn, Version version, Map<String,String> properties, DownloadListener... listeners) throws Exception {
+        if (!inited) {
+            init();
+        }
+
+        File returnVal = null;
+
+        final Entry<IMavenProjectFacade,MavenProject> entry = bsnMap.get(bsn);
+
+        if (entry != null) {
+            final MavenProject mavenProject = entry.getValue();
+
+            final File bundleFile = new File(mavenProject.getBuild().getDirectory() + "/" + mavenProject.getBuild().getFinalName() + ".jar");
+
+            if (!bundleFile.exists()) {
+                MavenUtil.execJarMojo(mavenProject, new NullProgressMonitor());
```

Comment:
Yes, but you will probably need to make a job to do the work since you aren't using async http client stuff from bnd.